model/pdata/traces/alias_trace.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "go.opentelemetry.io/collector/model/pdata/traces"
+
+// This file contains aliases for traces data structures.
+
+import "go.opentelemetry.io/collector/model/internal/pdata"
+
+// Marshaler is an alias for pdata.TracesMarshaler interface.
+type Marshaler = pdata.TracesMarshaler
+
+// Unmarshaler is an alias for pdata.TracesUnmarshaler interface.
+type Unmarshaler = pdata.TracesUnmarshaler
```

Comment:
If you do this kind of changes, you also need to change the funcs in the interface to not include "Traces". I think for that reason there was another proposal to use `ptraces` as package and keep the names exactly how they were.

Reply:
Do you mean to change `UnmarshalLogs` func in pdata.<strike>Traces</strike>Unmarshaler to `Unmarshal`? In that case how would unmarshalers implement all the functions to unmarshal logs, traces and metrics if they the same name?

Reply:
Do we need that?

Reply:
> Do we need that?

What do you mean?

Reply:
>  In that case how would unmarshalers implement all the functions to unmarshal logs, traces and metrics if they the same name?

Do we need this capability?

Reply:
I don't think it would be necessary to change `UnmarshalTraces` to `Unmarshal` as there would generally not be references to such implementing functions that start with `traces` and, if even if there were, they would not stutter due to the interposition of `Unmarshal` before `Traces`.  Indeed, the stdlib follows this practice with, e.g., [`json.Marshaler`](https://pkg.go.dev/encoding/json#Marshaler) having an `UnmarshalJSON` method.  As @dmitryax points out, this is the only way to support unmarshaling multiple types of values from a single implementation.

Reply:
> Do we need this capability?

Looking through the code, I don't see where we actually need the combined (un)marshalers. Probably we can just split  them. For example [here](https://github.com/open-telemetry/opentelemetry-collector/blob/9f2ae4ab8187d81475bcbbaa4d175c27d9bce9cd/model/otlp/pb_unmarshaler.go#L25) `pbUnmarshaler` becomes `pbLogsUnmarshaler` etc. @Aneurysm9 @bogdandrutu WDYT?

Reply:
I think that works for our uses, but I would still want to see the methods in the interfaces be `Unmarshal<Signal>`.  This provides flexibility for components that may support multiple signals and use these interfaces.  The [`awskinesisexporter` encoding system](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awskinesisexporter/internal/batch/encode.go#L48-L50) has an example of such a use.

Reply:
Makes sense. Then we probably don't need to split it